src/ImageViewerCaptureTool.cpp

```diff
@@ -21,94 +22,97 @@ ImageViewerCaptureTool::ImageViewerCaptureTool( double fovY, double fovX,
         height = width * tan(fovY * 0.5) / tan(fovX * 0.5);
     }
 
-    double aspectRatio = width * 1.0 / height;
+    setupViewer(width, height, fovY);
+}
 
-    initializeProperties(width, height);
-    _viewer->getCamera()->setComputeNearFarMode(osg::CullSettings::DO_NOT_COMPUTE_NEAR_FAR);
-    _viewer->getCamera()->setProjectionMatrixAsPerspective(fovY * 180.0 / M_PI, aspectRatio, 0.1, 1000);
+// create a RTT (render to texture) camera
+osg::Camera *ImageViewerCaptureTool::createRTTCamera(osg::Camera* cam, osg::Camera::BufferComponent buffer, osg::Texture2D *tex, osg::GraphicsContext *gfxc)
+{
+    osg::ref_ptr<osg::Camera> camera = cam;
+    camera->setClearColor(osg::Vec4(0, 0, 0, 1));
+    camera->setClearMask(GL_COLOR_BUFFER_BIT | GL_DEPTH_BUFFER_BIT);
+    camera->setRenderTargetImplementation(osg::Camera::FRAME_BUFFER_OBJECT);
+    camera->setRenderOrder(osg::Camera::PRE_RENDER, 0);
+    camera->setViewport(0, 0, tex->getTextureWidth(), tex->getTextureHeight());
+    camera->setGraphicsContext(gfxc);
+    camera->setDrawBuffer(GL_FRONT);
+    camera->setComputeNearFarMode(osg::CullSettings::DO_NOT_COMPUTE_NEAR_FAR);
+    camera->attach(buffer, tex);
+    return camera.release();
 }
 
-void ImageViewerCaptureTool::initializeProperties(uint width, uint height) {
-    _viewer = new osgViewer::Viewer;
+// create float textures to be rendered in FBO
+osg::Texture2D* ImageViewerCaptureTool::createFloatTexture(uint width, uint height)
+{
+    osg::ref_ptr<osg::Texture2D> tex2D = new osg::Texture2D;
+    tex2D->setTextureSize( width, height );
+    tex2D->setInternalFormat( GL_RGB32F_ARB );
+    tex2D->setSourceFormat( GL_RGBA );
+    tex2D->setSourceType( GL_FLOAT );
+    tex2D->setResizeNonPowerOfTwoHint( false );
+    tex2D->setFilter( osg::Texture2D::MIN_FILTER, osg::Texture2D::LINEAR );
+    tex2D->setFilter( osg::Texture2D::MAG_FILTER, osg::Texture2D::LINEAR );
+    return tex2D.release();
```

Comment:
why not to return a "osg::ref_ptr<osg::Texture2D>" ? Or just to create as a normal pointer?